### PR TITLE
added private exceptions

### PR DESCRIPTION
**Pull request by roxanatrufelea**
# Justification

Needed to handle the internal errors (5xx) 
# Implementation

Added private exceptions to handle the errors and send a suggestive error message to help the user

# Testing

used Rested with endpoint  http://localhost:8080/kitchen/users/ and Swagger

# Checklist

- [ ] I tested changes to product code in product
- [ ] I considered updates to the README
- [ ] I considered updates to the error code dictionary

### CODE REVIEW

File: iChef-WebApi/src/main/java/com/kitchen/iChef/Controller/ControllerAdvisor.java [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+    public ResponseEntity<String> handleMethodArgumentNotValidException(MethodArgumentNotValidException e) {
+        List<String> errorMessages = e.getBindingResult()
+                .getFieldErrors()
+                .stream()
+                .map(DefaultMessageSourceResolvable::getDefaultMessage)
+                .collect(Collectors.toList());
+        return new ResponseEntity<>(errorMessages.toString(), HttpStatus.BAD_REQUEST);
+    }
+
+
```
Comment (EnsyFane) [inline iChef-WebApi/src/main/java/com/kitchen/iChef/Controller/ControllerAdvisor.java:69]: Remove blank lines

File: iChef-WebApi/src/main/java/com/kitchen/iChef/Controller/Model/Request/AppUserRequest.java [hunk 2]
```diff
@@ -21,7 +21,7 @@
     private String lastName;
 
     @NotBlank(message = "The username of the user is not valid!")
-    @Size(min = 5)
+    @Size(min = 5, message = "The username length should be greater than 5")
```
Comment (EnsyFane) [inline iChef-WebApi/src/main/java/com/kitchen/iChef/Controller/Model/Request/AppUserRequest.java:24]: Isn't min inclusive? Meaning that the username should be greater than or equal to 5? Same applies to the resto of the messages

File: iChef-WebApi/src/main/java/com/kitchen/iChef/Exceptions/ValidationException.java [hunk 3]
```diff
@@ -0,0 +1,10 @@
+package com.kitchen.iChef.Exceptions;
+
+/***
+ * ValidationException should be thrown when the data received from a user are invalid
```
Comment (EnsyFane) [inline iChef-WebApi/src/main/java/com/kitchen/iChef/Exceptions/ValidationException.java:4]: *is* invalid

File: iChef-WebApi/src/main/java/com/kitchen/iChef/Service/UserService.java [hunk 4]
```diff
@@ -15,45 +20,74 @@
 
     private final UserRepository userRepository;
     private final TokenService tokenService;
+    private static final Logger LOGGER = LogManager.getLogger();
 
     @Autowired
     public UserService(UserRepository userRepository, TokenService tokenService) {
         this.userRepository = userRepository;
         this.tokenService = tokenService;
+        LOGGER.info("In Service");
```
Comment (EnsyFane) [inline iChef-WebApi/src/main/java/com/kitchen/iChef/Service/UserService.java:29]: Maybe log this as "User Service started"?

File: iChef-WebApi/src/main/java/com/kitchen/iChef/Service/UserService.java [hunk 5]
```diff
@@ -15,45 +20,74 @@
...
 
     public List<AppUser> getAllUsers() {
         return userRepository.findAll();
     }
 
     public AppUser deleteUser(String id) {
-        return userRepository.delete(id);
+        AppUser appUser;
+        try {
+
```
Comment (EnsyFane) [inline iChef-WebApi/src/main/java/com/kitchen/iChef/Service/UserService.java:56]: Remove unnecessary blank lines

**Review by EnsyFane — APPROVED** [review]

**Review by turcudaria — APPROVED** [review]

**Review by EnsyFane — APPROVED** [review]

**Review by turcudaria — APPROVED** [review]

**Review by Stefpark77 — APPROVED** [review]

**Pull request merged**